### PR TITLE
feat!: use standard image format

### PR DESCRIPTION
**Pull request by arunpoudel**
Looks like helm has a untold convention, [see](https://docs.renovatebot.com/modules/manager/helm-values/#additional-information)
on how images should be defined.

This PR will standardize how we define helm images in values.
This will allow renovate bot to automatically update our
helm values when a new image is published.

### CODE REVIEW

File: charts/coop-app-chart/values.yaml [hunk 1]
```diff
@@ -2,7 +2,11 @@ name: # the name of the service
 
 environment: # environment to which you deploy for ex production or staging
 
-image: # full image url of your image
+image:
+  repository: # the repository where the image is stored
+  tag: # the tag of the image
+  pullPolicy: Always # the policy to pull the image
```
Comment (AtzeDeVries) [inline charts/coop-app-chart/values.yaml:8]: i dont think we need to expose this as a user config.

File: charts/coop-app-chart/templates/_helpers.tpl [hunk 2]
```diff
@@ -41,7 +41,7 @@ helm.sh/chart: {{ include "coop-app-chart.chart" . }}
 {{ include "coop-app-chart.selectorLabels" . }}
 tags.datadoghq.com/env: {{ .Values.environment }}
 tags.datadoghq.com/service: {{ .Values.name }}
-tags.datadhoghq.com/version: {{ last (splitList ":" .Values.image) }}
+tags.datadhoghq.com/version: {{ last (splitList ":" .Values.image.tag) }}
```
Comment (AtzeDeVries) [inline charts/coop-app-chart/templates/_helpers.tpl:44]: we dont need to split anymore, we can just use the tag here.

File: charts/coop-app-chart/templates/_helpers.tpl [hunk 3]
```diff
@@ -30,7 +30,7 @@ Pod labels
...
```
Comment (AtzeDeVries) [inline charts/coop-app-chart/templates/_helpers.tpl:33]: we dont need to split anymore, we can just use the tag here.

File: charts/coop-app-chart/templates/deployment.yaml [hunk 4]
```diff
@@ -27,8 +27,8 @@ spec:
         - name: {{ .Values.name }}
           securityContext:
             {{- toYaml .Values.advanced.deployment.securityContext | nindent 12 }}
-          image: "{{ .Values.image }}"
-          imagePullPolicy: Always
+          image: "{{ required "A valid .Values.image.repository is required!" .Values.image.repository }}:{{ required "A valid .Values.image.tag is required!" .Values.image.tag }}"
```
Comment (AtzeDeVries) [inline charts/coop-app-chart/templates/deployment.yaml:30]: should also include registry, see comment below

File: charts/coop-app-chart/values.yaml [hunk 5]
```diff
@@ -2,7 +2,12 @@ name: # the name of the service
 
 environment: # environment to which you deploy for ex production or staging
 
-image: # full image url of your image
+image:
+  repository: # the repository where the image is stored
+  tag: # the tag of the image
+  registry: europe-docker.pkg.dev # the registry where the image is stored
+  pullPolicy: Always # the policy to pull the image
```
Comment (AtzeDeVries) [inline charts/coop-app-chart/values.yaml:9]:
```suggestion
```

Comment (arunpoudel) [reply charts/coop-app-chart/values.yaml:9]: Just wanted to sneak this in :P

File: charts/coop-app-chart/templates/_helpers.tpl [hunk 6]
```diff
@@ -30,7 +30,7 @@ Pod labels
 {{ include "coop-app-chart.selectorLabels" . }}
 tags.datadoghq.com/env: {{ .Values.environment }}
 tags.datadoghq.com/service: {{ .Values.name }}
-tags.datadoghq.com/version: {{ last (splitList ":" .Values.image) }}
+tags.datadhoghq.com/version: {{ .Values.image.tag }}
```
Comment (AtzeDeVries) [inline charts/coop-app-chart/templates/_helpers.tpl:33]: this still contains the typo

File: charts/coop-app-chart/templates/_helpers.tpl [hunk 7]
```diff
@@ -41,7 +41,7 @@ helm.sh/chart: {{ include "coop-app-chart.chart" . }}
 {{ include "coop-app-chart.selectorLabels" . }}
 tags.datadoghq.com/env: {{ .Values.environment }}
 tags.datadoghq.com/service: {{ .Values.name }}
-tags.datadhoghq.com/version: {{ last (splitList ":" .Values.image) }}
+tags.datadhoghq.com/version: {{ .Values.image.tag }}
```
Comment (AtzeDeVries) [inline charts/coop-app-chart/templates/_helpers.tpl:44]: this still contains the typo

**Review by AtzeDeVries — APPROVED** [review]

**Pull request merged**